x.py

```diff
@@ -4,6 +4,16 @@
 
 import os
 import sys
+
+# If this is python2, check if python3 is available and re-execute with that
+# interpreter.
+if sys.version_info.major < 3:
+    try:
+        os.execvp("python3", ["python3"] + sys.argv)
```

Comment:
Can we try to use the alternate forms (e.g., I take it Windows would prefer something like `py --version 3 ...` perhaps?

Reply:
I don't know how useful that would be, since `download-ci-llvm` currently only works on linux. I can add it just in case https://github.com/rust-lang/rust/issues/77084 ever gets fixed if you like.

Reply:
Also, I'm not able to test on windows, so I'm a little hesitant to add code that I haven't tried.